src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+      name: 'logo-dark.png'
+    }]
+  },{
+    name: 'Contents.json',
+    type: 'json'
+  }]
+}]} />
+
+If you want to dive deeper on asset catalogs I suggest this article: [Working with Assets Catalogs in Xcode](https://learnappmaking.com/xcode-assets-catalogs-how-to/).
+
+### Colors
+
+A colorset contains a single **Contents.json** file which describes the color and potentially different values for the color in different contexts like light and dark mode. Here is a color with light and dark mode:
+
+```json
+{
+  "colors": [
+    {
+      "idiom": "universal",
+      "color": {
+        "color-space": "srgb",
+        "components": {
+          "alpha": "1.0000",
+          "blue": "1.0000",
+          "red": "1.0000",
+          "green": "1.0000"
+        }
+      }
+    },
+    {
+      "idiom": "universal",
+      "color": {
+        "color-space": "srgb",
+        "components": {
+          "alpha": "1.0000",
+          "blue": "0.1255",
+          "red": "0.0863",
+          "green": "0.1255"
+        }
+      },
+      "appearances": [
+        {
+          "appearance": "luminosity",
+          "value": "dark"
+        }
+      ]
+    }
+  ],
+  "info": {
+    "author": "xcode",
+    "version": 1
+  }
+}
+```
+
+A few things to notice:
+
+- `colors` is an array of color objects because one color might have multiple values, one for each different context like light and dark mode.
+- A `color` can have an `appearances` array, which will let iOS know when that color should be used. There are 2 types of appearances you can have: luminosity and contrast. Luminosity can be 'light', 'dark', or 'any'. Contrast can be 'high' or nothing, meaning normal contrast. *(sidenote: this is kind of weird, from a user's perspective both contrast and luminosity are binary settings: light/dark and on/off for contrast. Why then are luminosity and contrast different for a developer?)*
+- `idiom` tells Xcode which device to target for that color. `universal` means all devices. You could also say `iPhone` `iPad` `Mac` and others.
+- `color-space` tells iOS which colorspace the color should be used in. The options are `srgb` and `display-p3`. If you wanted, you could provide different colors for different color spaces. Most of the time everything is assumed to be in the `srgb` colorspace, but if you want to learn more about colorspaces, here is a great article: [https://bjango.com/articles/colourmanagementgamut/](https://bjango.com/articles/colourmanagementgamut/)
+
+All of these settings multiply the number of potential different colors a single colorset could be. For example if you want to target light and dark mode as well as high and regular contrast you would have 4 (2 x 2) colors. If you added colorspace you would then have 8 (2 x 2 x 2). You don't need to necessarily have different colors for each, but you can see how complicated it can potentially get. Once you have your colors defined in an asset catalog you can reference them in code:
```

Comment:
You don't need to necessarily have different colors for each, but you can see how complicated it can potentially get. > You don’t need to have different colors for each, but you can see how complicated it can get.